browser/brave_stats/brave_stats_updater_browsertest.cc

```diff
@@ -76,187 +85,117 @@ class BraveStatsUpdaterBrowserTest : public InProcessBrowserTest {
   }
 
   void OnStandardStatsUpdated(const GURL& update_url) {
-    stats_standard_endpoint_was_called_ = true;
+    if (wait_for_standard_stats_updated_loop_) {
+      wait_for_standard_stats_updated_loop_->Quit();
+    }
+
+    on_standard_stats_updated_ = true;
+
     // We get //1/usage/brave-core here, so ignore the first slash.
     EXPECT_STREQ(update_url.path().c_str() + 1, "/1/usage/brave-core");
     update_url_ = update_url.spec();
-    wait_for_callback_loop_->Quit();
   }
 
   void WaitForStandardStatsUpdatedCallback() {
-    if (stats_standard_endpoint_was_called_)
+    if (wait_for_standard_stats_updated_loop_ || on_standard_stats_updated_) {
       return;
-    wait_for_callback_loop_.reset(new base::RunLoop);
-    wait_for_callback_loop_->Run();
+    }
+
+    wait_for_standard_stats_updated_loop_.reset(new base::RunLoop);
+    wait_for_standard_stats_updated_loop_->Run();
   }
 
   void OnThresholdStatsUpdated(const GURL& update_url) {
-    stats_threshold_endpoint_was_called_ = true;
+    if (wait_for_threshold_stats_updated_loop_) {
+      wait_for_threshold_stats_updated_loop_->Quit();
+    }
+
+    on_threshold_stats_updated_ = true;
+
     // We get //1/usage/brave-core-threshold here, so ignore the first slash.
     EXPECT_STREQ(update_url.path().c_str() + 1,
                  "/1/usage/brave-core-threshold");
     update_url_ = update_url.spec();
-    wait_for_callback_loop_->Quit();
   }
 
   void WaitForThresholdStatsUpdatedCallback() {
-    if (stats_threshold_endpoint_was_called_)
+    if (wait_for_threshold_stats_updated_loop_ || on_threshold_stats_updated_) {
       return;
-    wait_for_callback_loop_.reset(new base::RunLoop);
-    wait_for_callback_loop_->Run();
-  }
+    }
 
-  bool StandardStatsEndpointWasCalled() {
-    return stats_standard_endpoint_was_called_;
+    wait_for_threshold_stats_updated_loop_.reset(new base::RunLoop);
+    wait_for_threshold_stats_updated_loop_->Run();
   }
 
  private:
-  TestingPrefServiceSimple testing_local_state_;
-  std::unique_ptr<base::RunLoop> wait_for_callback_loop_;
-  bool stats_standard_endpoint_was_called_ = false;
-  bool stats_threshold_endpoint_was_called_ = false;
-  bool referral_was_initialized_ = false;
+  std::unique_ptr<base::RunLoop> wait_for_referral_initialized_loop_;
+  std::unique_ptr<base::RunLoop> wait_for_standard_stats_updated_loop_;
+  std::unique_ptr<base::RunLoop> wait_for_threshold_stats_updated_loop_;
+
+  std::string referral_code_;
   std::string update_url_;
+
+  bool on_referral_initialized_ = false;
+  bool on_standard_stats_updated_ = false;
+  bool on_threshold_stats_updated_ = false;
 };
 
 // Run the stats updater and verify that it sets the first check preference
 IN_PROC_BROWSER_TEST_F(BraveStatsUpdaterBrowserTest,
                        StatsUpdaterSetsFirstCheckPreference) {
-  // Ensure that first check preference is false
-  ASSERT_FALSE(GetLocalState()->GetBoolean(kFirstCheckMade));
```

Comment:
Would it make sense to keep this as soemthing like the following line?:
`EXPECT_FALSE(g_browser_process->local_state()->GetBoolean(kFirstCheckMade));`

Reply:
no, this runs too late in the process and there's no point at which a test has access to check it after g_browser_process has been initialized, but before the value is set